<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
"http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en">
<head>
<!-- 2022-01-23 Sun 15:36 -->
<meta http-equiv="Content-Type" content="text/html;charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1" />
<title>Accelerating Precision Medicine with Quantum Computing and AI</title>
<meta name="author" content="Shaurya Singh, Abrahim Mahmud" />
<meta name="generator" content="Org Mode" />
<style type="text/css">
<!--/*--><![CDATA[/*><!--*/
@charset "UTF-8";
@import url('https://fonts.googleapis.com/css?family=Roboto:300,350,400');
@import url('https://fonts.googleapis.com/css?family=Roboto Mono:300,340,380');
@import url('https://fonts.googleapis.com/css?family=Roboto Condensed:300,350,400');

body {
    font-family: "Roboto", sans;
    font-size: 16px;
    font-weight: 350;
    line-height: 1.3em;
}

#content {
    margin: 0 auto;
    max-width: 720px;
    margin-top: 5em;
    margin-bottom: 5em;
}

b {
    font-weight: 400;
}

a {
    color: #673AB7;
    text-decoration: none;
}

h1, h2, h3 {
    font-family: "Roboto Condensed", serif;
    font-weight: 400;
}

h1.title {
    font-weight: 400;
    line-height: 0.75em;
    padding-top: 1.em;
    padding-bottom: 0.5em;
}
h1 span.subtitle {
    font-size: 16px;
    font-weight: 300;
}
h2 {
    clear: both;
    padding-top: 1em;
    padding-bottom: 0em;
}


p {
    text-align: justify;
    text-justify: inter-word;
    hyphens: auto;
}

div.abstract {
    background: #f9f9ff;
    padding: .75em 1em .75em 1em;
    line-height: 1.25em;
    font-family: "Roboto Condensed", sans-serif;
    border: .5px solid #555599;
    padding-bottom: 0.25em;
}
div.abstract p {
    column-count: 1;
    margin: 0 0 0.5em 0;
}

div.figure {
    padding: 0.50em .00em 0.75em .00em;
}
div.figure p {
    column-count: 1;
    line-height: 1.2em;
    font-family: "Roboto Condensed", sans-serif;
}
div.sidefig::after {
    display: block;
    content: "";
    clear: both;
}
div.sidefig div.figure p:nth-of-type(1) {
    width: 60%;
    float: left;
}
div.sidefig div.figure p:nth-of-type(2) {
    font-size: 95%;
    /*    border-top: 1px solid black; */
    background-color: #f9f9ff;
    padding: 1em;
}
div.sidefig div.figure img {
    width: 100%;
}

div.figure img {
    width: 100%;
}
span.figure-number {
    font-weight: 500;
}

/* --- Footnote ------------------------------------------------------------ */
div.footpara {
    display: inline-block;
}
h2.footnotes {
    font-size: 16px;
    width: 50%;
    border-bottom: 1px solid black;
}
p.footpara {
    column-count: 1;
    line-height: 1.25em;
    font-family: "Roboto Condensed", sans-serif;
    margin: 0;
}
sup {
    font-size: 0.75em;
    vertical-align: top;
    position: relative; top: -0.5em;
}

/* --- Block quote --------------------------------------------------------- */
blockquote p {
    font-family: "Roboto Condensed", sans-serif;
    font-size: 95%;
    line-height: 1.15em;
    column-count: 1;
    padding-top: 1em;
    padding-bottom: 1.5em;
    border: .5px solid #f9f9ff;
    padding: 0.50em 1.00em 0.75em 1.00em;
    background-color: #f9f9ff;
    margin-bottom: 0.5em;
    border-radius: 5px;
}

/* --- Source blocks ------------------------------------------------------- */
code {
    font-family: "Roboto Mono", sans-serif;
    font-weight: 300;
}
pre.example::before {
    display: block;
    background-color: #fffff;
    content: "Output";
    font-family: "Roboto Mono", sans-serif;
    font-weight: 400;
    padding-bottom: 0.25em;
}

pre.example {
    font-size: 95%;
    background-color: #f9f9ff;
    line-height: 1.4em;
    font-family: "Roboto Mono", sans-serif;
    font-weight: 300;
    padding: 0.5em 1em 0.5em 1em;
}

pre.src {
    font-size: 95%;
    line-height: 1.4em;
    font-family: "Roboto Mono", sans-serif;
    font-weight: 300;
    padding: 0;
    margin: 0;
    overflow-x: visible;
    white-space: pre-wrap;
}
div.org-src-container {
    border: .5px solid #555599;
    padding: 0.50em 1.00em 0.75em 1.00em;
    background-color: #f9f9ff;
    margin-top: 0.25em;
    margin-bottom: 0.75em;
    border-radius: 5px;
}
label.org-src-name {
    display: block;
    font-family: "Roboto Mono", sans-serif;
    padding-bottom: .25em;
}
span.listing-number {
    font-weight: 500;
}
/*]]>*/-->
</style>
<script>
     MathJax = {
       chtml: {
         scale: 1
       },
       svg: {
         scale: 1,
         fontCache: "global"
       },
       tex: {
         tags: "ams",
         multlineWidth: "85%",
         tagSide: "right",
         tagIndent: ".8em"
       }
     };
     </script>
     <script id="MathJax-script" async
             src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-svg.js"></script>
</head>
<body>
<div id="content" class="content">
<h1 class="title">Accelerating Precision Medicine with Quantum Computing and AI
<br />
<span class="subtitle">Exploring the future of Personalized Health Care</span>
</h1>
<div id="table-of-contents" role="doc-toc">
<h2>Table of Contents</h2>
<div id="text-table-of-contents" role="doc-toc">
<ul>
<li><a href="#org7f00dc9">1. Introduction</a></li>
<li><a href="#org50dc8ab">2. AI in Precision Medicine</a>
<ul>
<li><a href="#org0f4ea44">2.1. Accelerating AI with Quantum Computing</a></li>
<li><a href="#org550b00e">2.2. Hybrid Quantum-Classical Neural Networks</a></li>
<li><a href="#orgf1dcc53">2.3. Generative Adversarial Networks</a>
<ul>
<li><a href="#org41abe31">2.3.1. MolGAN</a></li>
<li><a href="#org11fbd00">2.3.2. Qubit-efficient Quantum Molecule Generation</a></li>
</ul>
</li>
</ul>
</li>
<li><a href="#org8067a7f">3. Prototype</a>
<ul>
<li><a href="#orgdc1f33b">3.1. Image recognition (QuTorch-HG)</a></li>
<li><a href="#orgc9be14e">3.2. Quantum Accelerated Drug &amp; Molecule Generation (QGAN-HG)</a></li>
<li><a href="#orgde8011e">3.3. Data</a></li>
<li><a href="#orgdf68599">3.4. Tools and Hardware</a></li>
</ul>
</li>
<li><a href="#org9eb7645">4. Results</a>
<ul>
<li><a href="#org992b0fe">4.1. Speed</a></li>
<li><a href="#org5036164">4.2. Accuracy</a></li>
<li><a href="#org50eefd0">4.3. Pricing</a></li>
</ul>
</li>
<li><a href="#orgbf33c16">5. Conclusion and Further Research</a></li>
<li><a href="#org62beb51">6. References</a></li>
</ul>
</div>
</div>
<div class="abstract" id="org7303004">
<p>
The convergence of quantum computing, artificial intelligence, and precision medicine promises to revolutionize health care. Precision medicine methods identify phenotypes of patients with less-common responses to treatment or unique healthcare needs.  AI  leverages sophisticated computation and inference to generate insights, enables the system to reason and learn, and empowers clinician decision-making through augmented intelligence. Quantum-accelerated machine learning could support further breakthroughs in this area and ultimately enable causal inference models for drugs, identifying and explaining relationships among interventions and treatments on the one hand, and outcomes on the other in real-time, to provide the next-best medical action at the individual level.
</p>

<ul class="org-ul">
<li><b>Keywords:</b> Quantum Computing, Artificial intelligence, Precision Medicine</li>
<li><b>Environment:</b></li>
<li><b>License:</b> MIT</li>
</ul>

</div>


<div id="org80d1fed" class="figure">
<p><img src="./assets/sysone.jpg" alt="sysone.jpg" width="100%" />
</p>
<p><span class="figure-number">Figure 1: </span><b>IBM System One</b></p>
</div>

<blockquote>
<p>
&ldquo;By augmenting human performance, AI has the potential to markedly improve productivity, efficiency, workflow, accuracy and speed, both for [physicians] and for patients … What I’m most excited about is using the future to bring back the past: to restore the care in healthcare.&rdquo; . &#x2014; Eric Topol
</p>
</blockquote>

<div id="outline-container-org7f00dc9" class="outline-2">
<h2 id="org7f00dc9"><span class="section-number-2">1.</span> Introduction</h2>
<div class="outline-text-2" id="text-1">
<p>
Healthcare data—such as information from clinical trials, disease registries, electronic health records (EHRs), and medical devices—is growing at a compound annual growth rate of 36 percent (<a href="#citeproc_bib_item_13">NCHIT, 2021</a>). Increasingly, this data helps address 4 challenges associated with healthcare:
</p>

<ul class="org-ul">
<li>Better health</li>
<li>Lower cost</li>
<li>Enhanced patient experiences</li>
<li>Improved healthcare practitioner work lives.</li>
</ul>

<p>
At the same time, healthcare consumers are making more decisions and have to navigate an increasingly complex system. Traditionally, diagnosing a patient’s condition has been based heavily on the umbrella approach, which is based solely on the patient-reported symptoms, resulting in an umbrella diagnosis and treatment that frequently fails to achieve their intended effects due to individual variability. As a result, many existing therapies fail to achieve their intended effects due to individual variability. Precision medicine aims to allow tailor prevention and treatment with an individual approach. (<a href="#citeproc_bib_item_16">Ravizza et al., 2019</a>). Due to the complexity of human biology, individualized medicine requires taking into account aspects that go well beyond standard medical care. Medical care only has a relative contribution of 10 to 20 percent to outcomes; health-related behaviors, socioeconomic factors, and environmental aspects account for the other 80 to 90 percent (<a href="#citeproc_bib_item_10">Magnan &#38; , 2017</a>). Computationally, the interdependencies and correlations among these diverse contributors create formidable challenges concerning optimizing treatment effectiveness. (<a href="#citeproc_bib_item_11">McDermott et al., 2012</a>)
</p>

<p>
Significant investments are being made to deliver the right data and powerful insights at the point of care. Industry incumbents and new entrants alike are trying to create digital experiences that reinforce healthy, preventive behaviors. Despite that, accounting for the exponential possibilities from this diversity of new data is stretching the capabilities of classical computing systems. (<a href="#citeproc_bib_item_5">Grumbling &#38; Horowitz, 2020</a>).
</p>
</div>
</div>

<div id="outline-container-org50dc8ab" class="outline-2">
<h2 id="org50dc8ab"><span class="section-number-2">2.</span> AI in Precision Medicine</h2>
<div class="outline-text-2" id="text-2">

<div id="orgaf77c0f" class="figure">
<p><img src="./assets/precisionMedicineApproach.png" alt="precisionMedicineApproach.png" />
</p>
<p><span class="figure-number">Figure 2: </span><b>Precision Approach vs. Umbrella Approach</b></p>
</div>

<p>
A key aspect of tailoring medical approaches is proactivity. Early treatments and preventive interventions tend to drastically improve outcomes and optimize costs. Classical machine learning has already shown some promise in predicting the risk of future diseases for a range of patient groups based on EHRs (<a href="#citeproc_bib_item_6">Jauk et al., 2020</a>). Supervised and unsupervised quantum-enhanced machine learning techniques could allow earlier more accurate, and more granular risk predictions. Eventually, medical practitioners might even have the tools to understand how an individual’s risk for any given condition changes over time, enabled by continual virtual diagnostics based on ongoing data streams from individuals. (<a href="#citeproc_bib_item_7">Kumar et al., 2021</a>)
</p>
</div>

<div id="outline-container-org0f4ea44" class="outline-3">
<h3 id="org0f4ea44"><span class="section-number-3">2.1.</span> Accelerating AI with Quantum Computing</h3>
<div class="outline-text-3" id="text-2-1">
<p>
Quantum computers process information in a fundamentally different way from traditional computers. Previous computer technology advancements—such as integrated circuits—enabled faster computing, but were still based on classical information processing. These computers manipulate quantum bits (qubits). These are unlike classical bits, which store information as either a 0 or 1, and they can display uniquely quantum properties, such as entanglement. As a result, it becomes possible to construct quantum algorithms that can outperform their classical counterparts which are not able to leverage quantum phenomena. (<a href="#citeproc_bib_item_12">Montanaro, 2016</a>)
</p>

<ul class="org-ul">
<li>Chemistry, machine learning/artificial intelligence (AI), optimization, or simulation tasks. Machine learning has shown potential to be enhanced by quantum computing and is symbiotically helping drive quantum advances (<a href="#citeproc_bib_item_2">Carleo &#38; Troyer, 2017</a>)</li>
<li>Complex correlations and interdependencies among many highly interconnected elements, such as molecular structures in which many electrons interact</li>
<li>Inherent scaling limits of relevant classical algorithms. For instance, the resource requirements of classical algorithms may increase exponentially with problem size, as is the case when simulating the time evolution of quantum systems (<a href="#citeproc_bib_item_18">Whitfield et al., 2011</a>).</li>
</ul>
</div>
</div>

<div id="outline-container-org550b00e" class="outline-3">
<h3 id="org550b00e"><span class="section-number-3">2.2.</span> Hybrid Quantum-Classical Neural Networks</h3>
<div class="outline-text-3" id="text-2-2">

<div id="org4439770" class="figure">
<p><img src="./assets/hybridnetwork.png" alt="hybridnetwork.png" width="100%" />
</p>
<p><span class="figure-number">Figure 3: </span><b>Example of a Hybrid Quantum-Classical Neural Network</b></p>
</div>

<p>
The most computationally demanding part of gradient-based algorithms, namely, computing the fitness function and its gradient for control input, can be accomplished by the process of evolution and measurement on quantum hardware. By posing queries to and receiving answers from these devices, classical computing devices update the control parameters until an optimal control solution is found (<a href="#citeproc_bib_item_9">Li et al., 2017</a>). Using this hybrid approach gives rise to interesting areas of research that seek to leverage the principles of quantum mechanics to augment machine learning or vice-versa. Enabling us to enhance classical ML algorithms by outsourcing difficult calculations to a quantum computer.
</p>

<p>
To create a quantum-classical neural network, one can implement a hidden layer for a neural network using a parameterized quantum circuit, a quantum circuit where the rotation angles for each gate are specified by the components of a classical input vector. The outputs from the neural network&rsquo;s previous layer will be collected and used as the inputs for a parameterized circuit. The measurement statistics of the circuit can then be collected and used as inputs for the following layer.
</p>

<p>
Each classical equation is modeled through the nonlinear function
</p>

\begin{align}
h_{x}=\sigma(w_{a}x_{a}+w_{b}x_{b}+w_{c}x_{c})
\end{align}

<p>
where \(h_{i}\) is the value of neuron i at each hidden layer. \(R(h_{i})\) represents any rotation gate about an angle equal to \(h_{i}\) and \(y\) is the final prediction value generated from the hybrid network.
</p>


<div id="orgb85b796" class="figure">
<p><img src="./assets/neuralnetworkQC.png" alt="neuralnetworkQC.png" width="100%" />
</p>
<p><span class="figure-number">Figure 4: </span><b>Quantum-classical Neural Network using a Parameterized Quantum Circuit</b></p>
</div>
</div>
</div>

<div id="outline-container-orgf1dcc53" class="outline-3">
<h3 id="orgf1dcc53"><span class="section-number-3">2.3.</span> Generative Adversarial Networks</h3>
<div class="outline-text-3" id="text-2-3">
<p>
Generative Adversarial Networks, or GANs for short, are an approach to generative modeling using deep learning methods, such as convolutional neural networks. Generative modeling is an unsupervised learning task in machine learning that involves automatically discovering and learning the regularities or patterns in input data in such a way that the model can be used to generate or output new examples that plausibly could have been drawn from the original dataset (<a href="#citeproc_bib_item_3">Choi et al., 2020</a>)
</p>

<p>
GANs are a clever way of training a generative model by framing the problem as a supervised learning problem with two sub-models: the generator model that we train to generate new examples, and the discriminator model that tries to classify examples as either real (from the domain) or fake (generated). The two models are trained together in a zero-sum game, adversarial, until the discriminator model is fooled about half the time, meaning the generator model is generating plausible examples. This can be modeled through the equation (<a href="#citeproc_bib_item_17">Wang et al., 2020</a>)
</p>

\begin{align}
\min_{G}\max_{D}\mathbb{E}_{x\sim p_{\text{data}}(x)}[\log{D(x)}] +  \mathbb{E}_{z\sim p_{\text{generated}}(z)}[1 - \log{D(G(z))}]
\end{align}
</div>

<div id="outline-container-org41abe31" class="outline-4">
<h4 id="org41abe31"><span class="section-number-4">2.3.1.</span> MolGAN</h4>
<div class="outline-text-4" id="text-2-3-1">
<p>
Existing drug discovery pipelines take 5-10 years and cost billions of dollars. Computational approaches aim to sample from regions of the whole molecular and solid-state compounds called chemical space which could be on the order of 1060. Deep generative models can model the underlying probability distribution of both the physical structures and property of drugs and relate them nonlinearly. By exploiting patterns in massive datasets, these models can distill salient features that characterize the molecules. We can utilize Generative Adversarial Networks (GANs) discover drug candidates by generating molecular structures that obey chemical and physical properties and show affinity towards binding with the receptor for a target disease. (<a href="#citeproc_bib_item_8">Li et al., 2021</a>)
</p>

<p>
Currently, this is accomplished through the Tensorflow library <a href="https://github.com/nicola-decao/MolGAN">MolGAN</a>. However,  However, classical GANs cannot explore certain regions of the chemical space and suffer from curse-of-dimensionality. Computing these drug candidates can be computationally expensive, and the resource requirements of these classical algorithms increase exponentially with problem size. On the other hand. A full quantum GAN may require more than 90 qubits even to generate QM9-like small molecules, and is impractical in the current day and age
</p>
</div>
</div>

<div id="outline-container-org11fbd00" class="outline-4">
<h4 id="org11fbd00"><span class="section-number-4">2.3.2.</span> Qubit-efficient Quantum Molecule Generation</h4>
<div class="outline-text-4" id="text-2-3-2">
<p>
Once again, we can apply our hybrid approach. A qubit-efficient quantum GAN with a hybrid generator (QGAN-HG) can be used to learn a richer representation of molecules via searching exponentially large chemical space with fewer qubits and more efficiently than a classical GAN. The QGAN-HG model is composed of a hybrid quantum generator that supports various number of qubits and quantum circuit layers, and, a classical discriminator. The approach is significantly quicker than our classical GAN model.
</p>


<div id="org3d8645b" class="figure">
<p><img src="./assets/systemarchitecture.png" alt="systemarchitecture.png" />
</p>
</div>
</div>
</div>
</div>
</div>

<div id="outline-container-org8067a7f" class="outline-2">
<h2 id="org8067a7f"><span class="section-number-2">3.</span> Prototype</h2>
<div class="outline-text-2" id="text-3">
<p>
To test the speed, practicality, efficiency, and cost of quantum-accelerated ML as well as its usefulness in precision medicine, we have devised two prototypes written in Python. The concise, expressive, and dynamic nature of the Python language makes it well suited for prototyping tasks. Notebook one will test how viable our QML approach is at accelerating image and text sorting. This script can be adapted to identify mutations, distinguish genomic variants, as well as identify an indivudual&rsquo;s susceptiblity to rare diseases through an analysis of their previous health and history. Notebook 2 will generate viable drugs based on ones that currently exist, and will test how viable our QML approach is to accelerating current conventional drug discovery pipelines. Both models utilize PyTrorch and IBM&rsquo;s Quantum Services for training and testing
</p>

<p>
The full code for both notebooks are available under <a href="prototype/">the prototype folder</a>.
</p>
</div>

<div id="outline-container-orgdc1f33b" class="outline-3">
<h3 id="orgdc1f33b"><span class="section-number-3">3.1.</span> Image recognition (QuTorch-HG)</h3>
<div class="outline-text-3" id="text-3-1">

<div id="orgb18ed29" class="figure">
<p><img src="./assets/imagenet.png" alt="imagenet.png" width="100%" />
</p>
<p><span class="figure-number">Figure 5: </span><b>Hybrid Base Nueral Network</b></p>
</div>

<p>
We have created a base neural network model, which utilizes hybrid machine learning to create a model trained from any dataset in ImageNet format. The base model is used as the base for Transfer Learning, on an Image Classification task (based on resnet18). The last layer of this pre-trained model (fully-connected/fc layer) is then modified through a quantum machine learning framework, generating a new model. We will be testing its efficiency, practicality, and accuracy. We are training the model with 4 Qubits at 8 epochs. Loss is calculated using the Huber loss method, with the following equation
</p>

\begin{align}
L_{\delta}=
    \left\{\begin{matrix}
        \frac{1}{2}(y - \hat{y})^{2} & if \left | (y - \hat{y})  \right | < \delta\\
        \delta ((y - \hat{y}) - \frac1 2 \delta) & otherwise
    \end{matrix}\right.
\end{align}

<p>
Accuracy is calculated via a confusion matrix, with the following metrics
</p>

\begin{align}
Accuracy &= \frac{TP+TN}{TP+TN+FP+FN}\\
Precision &= \frac{TP}{TP+FP}\\
Recall &= \frac{TP}{TP+FN}\\
F1 &= \frac{2*Precision*Recall}{Precision+Recall} = \frac{2*TP}{2*TP+FP+FN}
\end{align}
</div>
</div>

<div id="outline-container-orgc9be14e" class="outline-3">
<h3 id="orgc9be14e"><span class="section-number-3">3.2.</span> Quantum Accelerated Drug &amp; Molecule Generation (QGAN-HG)</h3>
<div class="outline-text-3" id="text-3-2">

<div id="org9e45edd" class="figure">
<p><img src="./assets/quganflow.png" alt="quganflow.png" width="100%" />
</p>
<p><span class="figure-number">Figure 6: </span><b>Quantum Accelerated Molecule Generation</b></p>
</div>

<p>
Firstly, only generated molecules that have high affinity towards the receptor binding sites are considered as valid. (<a href="#citeproc_bib_item_1">Andreyev &#38; Zybailov, 2020</a>) Next, a parameterized quantum circuit with last-layer N measures the expectation values and a processes it through a classical stage. Then, we apply the atom layer and bond layer to generate synthetic molecular graphs. Afterwards, a batch of real molecules from a training dataset (in this case we are using QM9) and a batch of generated synthetic molecules are fed into a classical discriminator for real/synthetic prediction and Frechet distance score calculation.
</p>
</div>
</div>

<div id="outline-container-orgde8011e" class="outline-3">
<h3 id="orgde8011e"><span class="section-number-3">3.3.</span> Data</h3>
<div class="outline-text-3" id="text-3-3">
<p>
Our Image Data is obtained from Standford&rsquo;s ImageNet collection, a large-scale ontology of images built upon the WordNet structure. ImageNet aims to populate the majority of the 80,000 synsets of WordNet with an average of 500–1000 clean and full resolution images, with currently over 14,197,122 images and 21841 synsets indexed. The specific dataset used in this example can be found at <a href="https://www.kaggle.com/paultimothymooney/chest-xray-pneumonia">Paul Timothy: Chest X-RAY Pneumonia Dataset</a>, and is liscensed under CC0 1.0: Public Domain.
</p>

<p>
The Molecular data used to train our MolGAN and QuGAN models is the QM9 Dataset obtained from Anatole von Lilienfeld. The dataset contains the computed geometric, energetic, electronic, and thermodynamic properties for 134k stable small organic molecules made up of CHONF. These molecules correspond to the subset of all 133,885 species with up to nine heavy atoms (CONF) out of the GDB-17 chemical universe of 166 billion organic molecules. The model is trained on geometries, corresponding harmonic frequencies, dipole moments, polarizabilities, along with energies and enthalpies. (<a href="#citeproc_bib_item_15">Raghunathan Ramakrishnan et al., 2019</a>)
</p>
</div>
</div>

<div id="outline-container-orgdf68599" class="outline-3">
<h3 id="orgdf68599"><span class="section-number-3">3.4.</span> Tools and Hardware</h3>
<div class="outline-text-3" id="text-3-4">
<p>
The open-source Qiskit framework provides convenient access to multiple quantum simulators as well as a real quantum computer backend. The user can choose to utilize either IBM&rsquo;s cloud-based QASM simulator technology, Google&rsquo;s local equivalent Cirq, and Pennylane&rsquo;s quicker but less accurate lightning simulator. All three backends allow for quick training and testing via quantum simulators and real quantum hardware (<a href="#citeproc_bib_item_4">Cross, 2018</a>).
</p>


<div id="orgf6cffee" class="figure">
<p><img src="./assets/simulatedvsreal.png" alt="simulatedvsreal.png" width="100%" />
</p>
<p><span class="figure-number">Figure 7: </span><b>Actual vs. Simulated Hardware</b></p>
</div>

<p>
While simulated hardware produces <i>slightly</i> different results to actual hardware, the margin is within 1%. The tests below were conducted using actual IBMQ hardware, on the <code>ibmq_lima</code> quantum computer (<a href="#citeproc_bib_item_14">Quantum, 2021</a>)
</p>

<p>
Code is developed in Jupyter notebooks allowing for quick prototypng, and utilize PyTorch for pre-processing and post-processing of our neural network, taking advantage of GPU Acceleration via Nvidia CUDA if available. This allows us to process images in realtime on Google&rsquo;s Compute Engine VM&rsquo;s via Google Colab, allowing for low operating costs, high performance, and good portability.
</p>


<div id="org3b42495" class="figure">
<p><img src="./assets/circuit.png" alt="circuit.png" width="100%" />
</p>
<p><span class="figure-number">Figure 8: </span><b>Sample Generated Quantum Circuit on IBMQ</b></p>
</div>
</div>
</div>
</div>

<div id="outline-container-org9eb7645" class="outline-2">
<h2 id="org9eb7645"><span class="section-number-2">4.</span> Results</h2>
<div class="outline-text-2" id="text-4">
</div>
<div id="outline-container-org992b0fe" class="outline-3">
<h3 id="org992b0fe"><span class="section-number-3">4.1.</span> Speed</h3>
<div class="outline-text-3" id="text-4-1">
<p>
This is the largest benefit of quantum-accelerated machine learning.  We can see that in both algorithms, quantum computing provided an exponential increase in speed over the non-accelerated counterpart. In the case of QGAN, We can see anywhere from a 8-32% decrease in the time needed to generate molecules, with the same input parameters. In the case of our QuTorch-HG algorithm, it can process a batch of images within 1/10th of a second, allowing for models to be trained at 95% accuracy in under 5 minutes. A similar model, written with tensorflow and trained on the same cpu, took 32 minutes to achieve 94.3% accuracy.
</p>


<div id="org4852db6" class="figure">
<p><img src="./assets/quganresults.png" alt="quganresults.png" width="100%" />
</p>
<p><span class="figure-number">Figure 9: </span><b>Training comparison among GAN flavors</b></p>
</div>
</div>
</div>

<div id="outline-container-org5036164" class="outline-3">
<h3 id="org5036164"><span class="section-number-3">4.2.</span> Accuracy</h3>
<div class="outline-text-3" id="text-4-2">
<p>
In both prototypes, accuracy was as expected. The QuTorch-HG algorithm tested at around 96.23% accuracy on average on average after 8 epochs. The QGAN prototype created valid molecules 100% of the time during our testing.
</p>

<p width="100%">
<img src="./assets/output1.png" alt="output1.png" width="100%" />
<img src="./assets/output2.png" alt="output2.png" />
</p>


<div id="orgb8fe3e1" class="figure">
<p><img src="./assets/output3.png" alt="output3.png" width="100%" />
</p>
<p><span class="figure-number">Figure 10: </span><b>Analysis of Pneumonia through our QuTorch-HG algorithm</b></p>
</div>


<div id="orgad2eeeb" class="figure">
<p><img src="./assets/output4.png" alt="output4.png" width="100%" />
</p>
<p><span class="figure-number">Figure 11: </span><b>Sample Generated Molecule through our QGAN-HG algorithm</b></p>
</div>
</div>
</div>

<div id="outline-container-org50eefd0" class="outline-3">
<h3 id="org50eefd0"><span class="section-number-3">4.3.</span> Pricing</h3>
<div class="outline-text-3" id="text-4-3">
<p>
As of early 2022, IBM Quantum Services allows researchers and students to use their 5 qubit quantum computers for development free of charge. The GPU accelerator was provided by Google&rsquo;s Colab program, free of charge as well. Those looking for real-time analysis can utilize Google Compute Engine VM&rsquo;s, such as the A2 Accelerator for just $0.009 an hour. Our hybrid model is effecient and as all computation is handled through the cloud, energy costs are nominal.
</p>

<p>
On the other hand our algorithms can also be applied to healthcare cost analysis, such as improving insurance pricing computations, allowing for lower average premiums, as well as better-tailored premium options. We strongly believe investing in quantum computing now will result in increased profits in the future.
</p>
</div>
</div>
</div>

<div id="outline-container-orgbf33c16" class="outline-2">
<h2 id="orgbf33c16"><span class="section-number-2">5.</span> Conclusion and Further Research</h2>
<div class="outline-text-2" id="text-5">
<p>
Healthcare data, collected from the many different modalities and locations that surround patients—is underused today. Clinical factors only account for a small percentage of health-relevant data. Therefore, obtaining actionable insights from other areas, including genomics, behavior, and social and environmental influences, is critical.
</p>

<p>
Quantum-enhanced machine learning algorithms stand out for their breadth of application, their efficiency and relatively low running costs. We are entering an era where the characteristics of health datasets, such as being frequently heterogeneous and unevenly distributed, are producing complex computational challenges for modern AI. For instance, researchers have been exploring how to speed up the computationally expensive algorithms that are at the core of machine learning and AI modeling with quantum approaches, including operations with large matrices.
</p>

<p>
The maturity horizons of quantum computing use cases in healthcare depend not only on cross-industry quantum algorithm and hardware developments, but also on the increasing availability of relevant data. While certain modalities are already being leveraged for advanced computational modeling, health-relevant datasets are rapidly growing from increases in both the number of people being “instrumented,” as well as the data types being collected. As the number of accessible health- relevant data sources continues to grow, the potential for quantum computing to add value likewise increases.
</p>
</div>
</div>

<div id="outline-container-org62beb51" class="outline-2">
<h2 id="org62beb51"><span class="section-number-2">6.</span> References</h2>
<div class="outline-text-2" id="text-6">
<style>.csl-entry{text-indent: -1.5em; margin-left: 1.5em;}</style><div class="csl-bib-body">
  <div class="csl-entry"><a id="citeproc_bib_item_1"></a>Andreyev, D. S., &#38; Zybailov, B. L. (2020). Integration of flow cytometry and single cell sequencing. <i>Trends in Biotechnology</i>, <i>38</i>(2), 133–136. <a href="https://doi.org/10.1016/j.tibtech.2019.09.002">https://doi.org/10.1016/j.tibtech.2019.09.002</a></div>
  <div class="csl-entry"><a id="citeproc_bib_item_2"></a>Carleo, G., &#38; Troyer, M. (2017). Solving the quantum many-body problem with artificial neural networks. <i>Science</i>, <i>355</i>(6325), 602–606. <a href="https://doi.org/10.1126/science.aag2302">https://doi.org/10.1126/science.aag2302</a></div>
  <div class="csl-entry"><a id="citeproc_bib_item_3"></a>Choi, K., Grover, A., Singh, T., Shu, R., &#38; Ermon, S. (2020). Fair generative modeling via weak supervision. In H. D. III &#38; A. Singh (Eds.), <i>Proceedings of the 37th international conference on machine learning</i> (Vol. 119, pp. 1887–1898). PMLR. <a href="https://proceedings.mlr.press/v119/choi20a.html">https://proceedings.mlr.press/v119/choi20a.html</a></div>
  <div class="csl-entry"><a id="citeproc_bib_item_4"></a>Cross, A. (2018). The ibm q experience and qiskit open-source quantum computing software. <i>Aps March Meeting Abstracts</i>, <i>2018</i>, L58–003.</div>
  <div class="csl-entry"><a id="citeproc_bib_item_5"></a>Grumbling, E., &#38; Horowitz, M. (2020). <i>Quantum computing</i>. <a href="https://doi.org/10.17226/25196">https://doi.org/10.17226/25196</a></div>
  <div class="csl-entry"><a id="citeproc_bib_item_6"></a>Jauk, S., Kramer, D., Großauer, B., Rienmüller, S., Avian, A., Berghold, A., Leodolter, W., &#38; Schulz, S. (2020). Risk prediction of delirium in hospitalized patients using machine learning: An implementation and prospective evaluation study. <i>Journal of the American Medical Informatics Association</i>, <i>27</i>(9), 1383–1392. <a href="https://doi.org/10.1093/jamia/ocaa113">https://doi.org/10.1093/jamia/ocaa113</a></div>
  <div class="csl-entry"><a id="citeproc_bib_item_7"></a>Kumar, Y., Koul, A., Sisodia, P. S., Shafi, J., Verma, K., Gheisari, M., &#38; Davoodi, M. B. (2021). Heart failure detection using quantum-enhanced machine learning and traditional machine learning techniques for internet of artificially intelligent medical things. <i>Wireless Communications and Mobile Computing</i>, <i>2021</i>, 1–16. <a href="https://doi.org/10.1155/2021/1616725">https://doi.org/10.1155/2021/1616725</a></div>
  <div class="csl-entry"><a id="citeproc_bib_item_8"></a>Li, J., Topaloglu, R., &#38; Ghosh, S. (2021). Quantum generative models for small molecule drug discovery. <i>Arxiv:2101.03438 [Quant-Ph]</i>. <a href="http://arxiv.org/abs/2101.03438">http://arxiv.org/abs/2101.03438</a></div>
  <div class="csl-entry"><a id="citeproc_bib_item_9"></a>Li, J., Yang, X., Peng, X., &#38; Sun, C.-P. (2017). Hybrid quantum-classical approach to quantum optimal control. <i>Physical Review Letters</i>, <i>118</i>(15). <a href="https://doi.org/10.1103/physrevlett.118.150503">https://doi.org/10.1103/physrevlett.118.150503</a></div>
  <div class="csl-entry"><a id="citeproc_bib_item_10"></a>Magnan, S., &#38; . (2017). Social determinants of health 101 for health care: Five plus five. <i>NAM Perspectives</i>, <i>7</i>(10). <a href="https://doi.org/10.31478/201710c">https://doi.org/10.31478/201710c</a></div>
  <div class="csl-entry"><a id="citeproc_bib_item_11"></a>McDermott, J. E., Wang, J., Mitchell, H., Webb-Robertson, B.-J., Hafen, R., Ramey, J., &#38; Rodland, K. D. (2012). Challenges in biomarker discovery: combining expert insights with statistical analysis of complex omics data. <i>Expert Opinion on Medical Diagnostics</i>, <i>7</i>(1), 37–51. <a href="https://doi.org/10.1517/17530059.2012.718329">https://doi.org/10.1517/17530059.2012.718329</a></div>
  <div class="csl-entry"><a id="citeproc_bib_item_12"></a>Montanaro, A. (2016). Quantum algorithms: an overview. <i>Npj Quantum Information</i>, <i>2</i>(1). <a href="https://doi.org/10.1038/npjqi.2015.23">https://doi.org/10.1038/npjqi.2015.23</a></div>
  <div class="csl-entry"><a id="citeproc_bib_item_13"></a>NCHIT. (2021). <i>Non-federal acute care hospital electronic health record adoption</i>. <a href="https://www.healthit.gov/data/">https://www.healthit.gov/data/</a></div>
  <div class="csl-entry"><a id="citeproc_bib_item_14"></a>Quantum, I. (2021). <i>Ibm quantum services</i>. <a href="https://quantum-computing.ibm.com/">https://quantum-computing.ibm.com/</a></div>
  <div class="csl-entry"><a id="citeproc_bib_item_15"></a>Raghunathan Ramakrishnan, Dral, P., Rupp, M., &#38; O. Anatole Von Lilienfeld. (2019). <i>Quantum chemistry structures and properties of 134 kilo molecules</i>. Figshare. <a href="https://doi.org/10.6084/M9.FIGSHARE.C.978904.V5">https://doi.org/10.6084/M9.FIGSHARE.C.978904.V5</a></div>
  <div class="csl-entry"><a id="citeproc_bib_item_16"></a>Ravizza, S., Huschto, T., Adamov, A., Böhm, L., Büsser, A., Flöther, F. F., Hinzmann, R., König, H., McAhren, S. M., Robertson, D. H., Schleyer, T., Schneidinger, B., &#38; Petrich, W. (2019). Predicting the early risk of chronic kidney disease in patients with diabetes using real-world data. <i>Nature Medicine</i>, <i>25</i>(1), 57–59. <a href="https://doi.org/10.1038/s41591-018-0239-8">https://doi.org/10.1038/s41591-018-0239-8</a></div>
  <div class="csl-entry"><a id="citeproc_bib_item_17"></a>Wang, X., Han, Y., Leung, V. C. M., Niyato, D., Yan, X., &#38; Chen, X. (2020). Convergence of edge computing and deep learning: A comprehensive survey. <i>IEEE Communications Surveys &#38; Tutorials</i>, <i>22</i>(2), 869–904. <a href="https://doi.org/10.1109/comst.2020.2970550">https://doi.org/10.1109/comst.2020.2970550</a></div>
  <div class="csl-entry"><a id="citeproc_bib_item_18"></a>Whitfield, J. D., Biamonte, J., &#38; Aspuru-Guzik, A. (2011). Simulation of electronic structure hamiltonians using quantum computers. <i>Molecular Physics</i>, <i>109</i>(5), 735–750. <a href="https://doi.org/10.1080/00268976.2011.552441">https://doi.org/10.1080/00268976.2011.552441</a></div>
</div>

<blockquote>
<p>
We acknowledge the use of IBM Quantum services for this work. The views expressed are those of the authors, and do not reflect the official policy or position of IBM or the IBM Quantum team.
</p>
</blockquote>
</div>
</div>
</div>
</body>
</html>
